### PR TITLE
Droth 268 new asset type huoltotie

### PR DESCRIPTION
**Pull request by marquesfm**

### CODE REVIEW

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/LinearAssetService.scala [hunk 1]
```diff
@@ -86,16 +87,26 @@ trait LinearAssetOperations {
...
+
+    var roadLinks: Seq[RoadLink] = null
+    if (typeId == LinearAssetTypes.MaintenanceRoadAssetTypeId) {
+      roadLinks = roadLinksExist.filter(_.functionalClass > 4)
+    } else{
+      roadLinks = roadLinksExist
+    }
     val linkIds = roadLinks.map(_.linkId)
     val removedLinkIds = LinearAssetUtils.deletedRoadLinkIds(changes, roadLinks)
+    val removedLinkIdsWithFilterClass = LinearAssetUtils.deletedRoadLinkIds(changes, roadLinks.filter(_.functionalClass == 0))
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/LinearAssetService.scala]: This line is not needed anymore (removedLinkIdsWithFilterClass)

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/AssetPropertyService.scala [hunk 2]
```diff
@@ -26,6 +26,12 @@ class AssetPropertyService(eventbus: DigiroadEventBus, userProvider: UserProvide
     }
   }
 
+//  def getProperties(assetTypeId: Long): Seq[Property] = {
+//    databaseTransaction.withDynTransaction{
+//      Queries.availableProperties(assetTypeId)
+//    }
+//  }
+
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/AssetPropertyService.scala]: This comments should be deleted

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/oracle/OracleLinearAssetDao.scala [hunk 3]
```diff
@@ -401,6 +401,115 @@ class OracleLinearAssetDao(val vvhClient: VVHClient) {
...
 
+  /**
+    * Iterates a set of link ids with MaintenanceRoad asset type id and floating flag and returns linear assets. Used by LinearAssetService.getByRoadLinks
+    */
+  def fetchMaintenancesByLinkIds(maintenanceRoadAssetTypeId: Int, linkIds: Seq[Long], includeFloating: Boolean = false): Seq[PersistedLinearAsset] = {
+    val floatingFilter = if (includeFloating) "" else "and a.floating = 0"
+
+    val assets = MassQuery.withIds(linkIds.toSet) { idTableName =>
+      sql"""
+         select a.id, t.id, t.property_id, t.value_fi, p.property_type, p.public_id, p.required, pos.link_id,
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/oracle/OracleLinearAssetDao.scala]: the p.required field could be removed

File: UI/src/view/PiecewiseLinearAssetFormElements.js [hunk 4]
```diff
@@ -197,11 +256,98 @@
...
+      }).join('');
+
+      var accessRightsTemplate = template({className: className, optionTags: accessRightsTag, disabled: disabled, label: 'Käyttöoikeus', id: 'huoltotie_kayttooikeus'});
+      var maintenanceResponsibilityTemplate = template({className: className, optionTags: maintenanceResponsibilityTag, disabled: disabled, label: 'Huoltovastuu', id: 'huoltotie_huoltovastuu'});
+
+      return '<form class="input-unit-combination form-horizontal ' + className +'">'+accessRightsTemplate.concat(maintenanceResponsibilityTemplate)+textBoxValues+'</form>';
+    }
+
+      function inputElementValue(input) {
+          var things = _.map(input, function (ele) {
```
Comment (marquesfm) [inline UI/src/view/PiecewiseLinearAssetFormElements.js]: Maybe the variables "things" and "ele" need to be refactored.

File: digiroad2-api-oth/src/test/scala/fi/liikennevirasto/digiroad2/Digiroad2ApiSpec.scala [hunk 5]
```diff
@@ -133,7 +133,7 @@ class Digiroad2ApiSpec extends AuthenticatedApiSpec with BeforeAndAfter {
   test("get enumerated property values", Tag("db")) {
     getWithUserAuth("/enumeratedPropertyValues/10") {
       status should equal(200)
-      parse(body).extract[List[EnumeratedPropertyValue]].size should be(14)
+      parse(body).extract[List[EnumeratedPropertyValue]].size should be(13)
```
Comment (tvenhola) [inline digiroad2-api-oth/src/test/scala/fi/liikennevirasto/digiroad2/Digiroad2ApiSpec.scala]: Why did the number of enumerated property values decrease by one?

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/LinearAssetService.scala [hunk 6]
```diff
@@ -86,7 +87,14 @@ trait LinearAssetOperations {
     getByRoadLinks(typeId, roadLinks, change)
   }
 
-  private def getByRoadLinks(typeId: Int, roadLinks: Seq[RoadLink], changes: Seq[ChangeInfo]): Seq[PieceWiseLinearAsset] = {
+  private def getByRoadLinks(typeId: Int, roadLinksExist: Seq[RoadLink], changes: Seq[ChangeInfo]): Seq[PieceWiseLinearAsset] = {
+
+    var roadLinks: Seq[RoadLink] = null
```
Comment (tvenhola) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/LinearAssetService.scala]: Avoid using var, use val instead:
val roadLinks = roadLinksExist.filter(rl => rl.functionalClass > 4 || typeId == ...)

**Comment by tvenhola** [comment]
Was there integration API for this asset?

**Comment by halied** [comment]
Columns missing from read-only form, otherwise ok

File: digiroad2-api-oth/src/test/scala/fi/liikennevirasto/digiroad2/Digiroad2ApiSpec.scala (hunk 5 above)
Comment (MAC-CGI) [inline digiroad2-api-oth/src/test/scala/fi/liikennevirasto/digiroad2/Digiroad2ApiSpec.scala]: The query in "queries.scala" file was wronged, returns all type_id, not only the type_id = 10. When we include the new ones we saw that the query improperly returns 16.. We correct the query and the test above.

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/LinearAssetService.scala (hunk 6 above)
Comment (marquesfm) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/LinearAssetService.scala]: If we do like you say, for others assets type, the roadlinks return 0.
Or I don't see the point correctly?

Comment (marquesfm) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/LinearAssetService.scala]: Maybe if we do like this is better:
    val roadLinks = typeId match {
      case LinearAssetTypes.MaintenanceRoadAssetTypeId => roadLinksExist.filter(_.functionalClass > 4)
      case _ => roadLinksExist
    }

**Comment by tvenhola** [comment]
Tests passed after my last commit, can you verify that the functionality didn't change?

**Comment by marquesfm** [comment]
Same for me, all tests passed.
The functionality didn't change, all works fine.

**Comment by joanacoelho** [comment]
Ok for me

**Comment by MAC-CGI** [comment]
It seems OK for me

**Pull request merged**